### ISSUE TITLE
Auto-fix for `SIM108` breaks `if sys.version_info ...` mypy convention

### ISSUE THREAD

**Issue by edgarrmondragon** [charliermarsh/ruff#1753]
```python
import random
import sys


def _get_random_bytes(n: int):
    return random.getrandbits(n * 8).to_bytes(n, "little")

if sys.version_info >= (3, 9):
    randbytes = random.randbytes
else:
    randbytes = _get_random_bytes
```

From [the Mypy docs](https://mypy.readthedocs.io/en/stable/common_issues.html#python-version-and-system-platform-checks):

<blockquote>

Mypy supports the ability to perform Python version checks and platform checks (e.g. Windows vs Posix), ignoring code paths that won’t be run on the targeted Python version or platform. This allows you to more effectively typecheck code that supports multiple versions of Python or multiple operating systems.

More specifically, mypy will understand the use of [sys.version_info](https://docs.python.org/3/library/sys.html#sys.version_info) and [sys.platform](https://docs.python.org/3/library/sys.html#sys.platform) checks within if/elif/else statements.

</blockquote>

so this works well with `mypy`:

```console
$ mypy ternary_sys.py --python-version 3.8
Success: no issues found in 1 source file
```

But the ternary operation created by the `SIM108` autofix breaks the above convention:

```console
$ ruff --version
ruff 0.0.216

$ ruff --select SIM108 ternary_sys.py
ternary_sys.py:8:1: SIM108 Use ternary operator `randbytes = random.randbytes if sys.version_info >= (3, 9) else _get_random_bytes` instead of if-else-block
Found 1 error(s).
1 potentially fixable with the --fix option.

$ ruff --select SIM108 ternary_sys.py --fix
Found 1 error(s) (1 fixed, 0 remaining).

$ mypy ternary_sys.py --python-version 3.8
ternary_sys.py:8: error: Module has no attribute "randbytes"  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
```

**Comment by edgarrmondragon** [comment]
Maybe the autofix should check if `sys.version_info` or `sys.platform` are referenced

**Comment by charliermarsh** [comment]
Sounds good, we can definitely special-case it.

**Comment by charliermarsh** [comment]
We already special-case “if name = main” so there’s precedent.

**Comment by charliermarsh** [comment]
I can probably get to this tonight unless you want to @edgarrmondragon.
